src/construction.cpp

```diff
@@ -1500,3 +1500,24 @@ std::vector<std::string> construction::get_folded_time_string( int width ) const
     std::vector<std::string> folded_time = foldstring( time_text, width );
     return folded_time;
 }
+
+void finalize_constructions()
+{
+    for( construction &con : constructions ) {
+        if( con.post_special == &construct::done_vehicle ) {
+            std::set<itype_id> frames;
+            for( auto &vp : vpart_info::get_all() ) {
+                if( !vp->has_flag( "INITIAL_PART" ) ) {
+                    continue;
+                }
+                frames.insert( vp->item );
+            }
+
+            auto &comp = con.requirements.get_components();
+            comp.resize( 1 );
+            for( auto &frame : frames ) {
```

Comment:
You could rearrange the loops and get rid of the intermediate set, or at least reserve space in the vector since you know its final size here.